calculator_program.c

```diff
@@ -0,0 +1,143 @@
+#include<stdio.h>
+#include <stdbool.h> 
+#include<stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+bool checkValid(char *str){
+    int i=0;
+    while(str[i] != '\0'){
+        if(str[i]!='+' && str[i]!='-' && str[i]!='*' && str[i]!='/' && str[i]!=' ' && !isdigit(str[i])){
+            return false;
+        }
+        i++;
+    }
+
+    return true;
+}
+
+int priority(char ch){
+    if(ch=='+' || ch=='-'){
+        return 1;
+    }
+
+    if(ch=='*' || ch=='/'){
+        return 2;
+    }
+
+    return 0;//for any other symbol
+}
+
+int calculate(int num1,int num2,char ch){
+    if(ch=='+'){
+        return num1+num2;
+    }else if(ch=='-'){
+        return num1-num2;
+    }else if(ch=='*'){
+        return num1*num2;
+    }else if(ch=='/'){
+        if(num2==0){
+            printf("Error: Division by zero.");
+            exit(1);
+        }else{
+            return num1/num2;
+        }
+
+    }else{
+        printf("Error: Invalid Operator");
+        exit(1);
+    }
+}
+
+int solve(char *str){
```

Comment:
Too much of logic is in single function.
Make separate functions and use them here.